docs/src/adr/002adr-types.md

```diff
@@ -57,30 +98,49 @@ Z3](https://github.com/Z3Prover/z3), a type constant can be understood as an
 uninterpreted sort in SMT. Essentially, values of an uninterpreted type can
 be only checked for equality.
 
-Assume that notation `e <: T` means that an expression `e` has type `T`.
-(More precisely, `T` is a supertype of the type of `e`.)
-The following examples demonstrate the use of the type grammar:
+Like in programming languages, a type may contain a reference to a type alias.
+This is purely a convenience feature to make type annotations more concise and
+easier to maintain. We expect only the user to write type aliases, whereas
+tools should always exchange with types in the alias-free form.
 
-* `x` is an integer: `x <: "Int"`.
-* `f` is a function from an integer to an integer: `f <: "Int -> Int"`.
-* `f` is a function from a set of integers to a set of integers:
-    `f <: "Set(Int) -> Set(Int)"`.
+**Examples.**
+
+* `x` is an integer. Its type is `Int`.
+* `f` is a function from an integer to an integer. Its type is `Int -> Int`.
+* `f` is a function from a set of integers to a set of integers.
+    Its type is `Set(Int) -> Set(Int)`.
 * `r` is a record that has the fields `a` and `b`, where `a` is an integer
-    and `b` is a string: `r <: "[a: Int, b: Str]"`.
-* `F` is a set of functions from a tuple of integers to an integer:
-    `F <: "Set(<<Int, Int>> -> Int)"`.
-* `Foo` is an operator of an integer and of a string that returns an integer:
-    `Foo <: "(Int, Str) => Int"`.
+    and `b` is a string. Its type is `[a: Int, b: Str]`.
+* `F` is a set of functions from a pair of integers to an integer.
+   Its type is `Set(<<Int, Int>> -> Int)`.
+* `Foo` is an operator of an integer and of a string that returns an integer.
+   Its type is  `(Int, Str) => Int`.
 * `Bar` is a higher-order operator that takes an operator that takes
-    an integer and a string and returns an integer, and returns a Boolean:
-    `Bar <: "((Int, Str) => Int) => Bool"`.
+    an integer and a string and returns an integer, and returns a Boolean.
+   Its type is  `((Int, Str) => Int) => Bool`.
 * `Baz` is a polymorphic operator that takes two arguments of the same type
-    and returns a value of the type equal to the types of its arguments:
-    `Baz <: "(a, a) => a"`.
-* `Proc` and `Faulty` are sets of the same type:
-    `Proc <: "Set(PID)"` and `Faulty <: "Set(PID)"`.
+    and returns a value of the type equal to the types of its arguments.
+   Its type is `(a, a) => a`.
+* `Proc` and `Faulty` are sets of the same type.
+   Their type is `Set(PID)`.
+
+### 1.2. Type aliases
 
-### 1.2. Discussion
+We extend the type grammar of `T` with one more rule that introduces a type
+alias:
```

Comment:
```suggestion
We extend the type grammar of `T` with one more rule that introduces type
aliases:
```

Reply:
This intro to the section doesn't make sense to me, because the type alias is specified in the grammar to begin with, not added as an extensions here (the way the current document is written). Perhaps just say "kthe grammer of `T` includes a rule" rather than "we extend"?